### PR TITLE
Update Cilium implementation details for Cilium 1.14

### PR DESCRIPTION
**Pull request by youngnick**
/kind documentation

Updates Cilium's implementation details for the most recent release.

### CODE REVIEW

**Review by shaneutt — APPROVED** [review]
/approve
/lgtm

File: site-src/implementations.md [hunk 1]
```diff
@@ -132,8 +132,8 @@ solution for Kubernetes and other networking environments. It includes [Cilium
 Service Mesh][cilium-service-mesh], a highly efficient mesh data plane that can
 be run in [sidecarless mode][cilium-sidecarless] to dramatically improve
 performance, and avoid the operational complexity of sidecars. Cilium also
-supports the sidecar proxy model, offering choice to users. As of [Cilium 1.13][cilium113blog],
-Cilium supports Gateway API, passing conformance for v0.5.1.
+supports the sidecar proxy model, offering choice to users. As of [Cilium 1.14][cilium113blog],
```
Comment (thiagogcm) [inline site-src/implementations.md:135]: change to [cilium114blog] too

Comment (youngnick) [reply site-src/implementations.md]: 🤦‍♂️ Thanks!

**Pull request merged**